stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1346,6 +1364,7 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
     /// - parameter data: The replacement data.
     @inline(__always)
     public mutating func replaceSubrange(_ subrange: Range<Index>, with data: Data) {
+        if !(subrange.lowerBound == endIndex && subrange.count == 0) { _validateRange(subrange) }
```

Comment:
this mimics the same behavior as Array

`print(["hello", "world"][100..<100])`